.github/workflows/release.yml

```diff
@@ -0,0 +1,30 @@
+name: Release New Version
+
+on:
+  push:
+    branches: [ master ]
+
+jobs:
+  pipeline:
+    name: Pipeline
+    runs-on: ubuntu-latest
+    steps:
+      - name: Create new GitHub tag
+          id: new_tag
+          uses: butlerlogic/action-autotag@stable
+          with:
+            GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+            strategy: regex
+            root: "version"
+            regex_pattern: "^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$"
+            tag_prefix: "v"
+      - name: Create GitHub release
+          if: github.ref == 'refs/heads/master'
```

Comment:
we probably don't need this if statement if the workflow is only triggered by pushes to master, correct?

Reply:
Correct.